app/controllers/admin/redact_files_controller.rb

```diff
@@ -1,30 +1,45 @@
 module Admin
   class RedactFilesController < Admin::ApplicationController
     def new
-      @source_file = SourceFile.find(params[:source_file_id])
+      @record = if Flipper.enabled?(:show_imports_in_administrate)
+        Imports::Pdf.find(params[:import_id])
+      else
+        SourceFile.find(params[:source_file_id])
+      end
       authorize :redact_file, :new?
+
       # Use a custom layout
       render layout: "admin/pdf_editor"
     end
 
     def create
       respond_to do |format|
         format.json do
-          @source_file = SourceFile.find(params[:source_file_id])
+          @record = if Flipper.enabled?(:show_imports_in_administrate)
+            Import.find(params[:import_id])
+          else
+            SourceFile.find(params[:source_file_id])
+          end
           authorize :redact_file, :new?
+
           if params[:redacted_file]
-            @source_file.redacted_source_file.attach(params[:redacted_file])
-            @source_file.associated_occupation_standards.each do |occupation_standard|
+            if Flipper.enabled?(:show_imports_in_administrate)
+              @record.redacted_pdf.attach(params[:redacted_file])
+            else
+              @record.redacted_source_file.attach(params[:redacted_file])
+            end
+            @record.associated_occupation_standards.each do |occupation_standard|
               occupation_standard.redacted_document.attach(params[:redacted_file])
             end
-            @source_file.update(redacted_at: Time.current)
+            @record.update(redacted_at: Time.current)
+
             render json: {
               message: "Redacted document saved for all occupation standards associated to this source file",
               status: :ok
             }.to_json
           else
             render json: {
-              error: "Redacted document saved for all occupation standards assoaciated to this source file",
+              error: "Redacted document saved for all occupation standards associated to this source file",
```

Comment:
@FerPerales I just copied this over from SourceFile, but is this error statement correct? Seems like it should say that no redacted file was able to be attached?

Reply:
@littleforest oops! You are right. The original error message is incorrect

Reply:
fixed in e657003a585bd117f286e67aa6519f5029c4a0ba